internal/apps/user/projector.go

```diff
@@ -68,9 +67,34 @@ func GetUserName(ds projector.Datastore, uid int) (string, error) {
 		return user.Username, nil
 	}
 
-	if user.Level != "" {
-		parts = append(parts, fmt.Sprintf("(%s)", user.Level))
+	return strings.Join(parts, " "), nil
+}
+
+// GetUserLevel returns the structre level for user id.
+func GetUserLevel(ds projector.Datastore, uid int) (string, error) {
+	var user struct {
+		Level string `json:"structure_level"`
 	}
 
-	return strings.Join(parts, " "), nil
+	if err := ds.Get("users/user", uid, &user); err != nil {
+		return "", projector.NewClientError("users/user with id %d does not exist", uid)
+	}
```

Comment:
This part is now essentially duplicated. I don't really know go. Perhaps it was possible to only ask one time instead of requesting the same user twice